pandas/core/indexes/datetimelike.py

```diff
@@ -629,6 +651,28 @@ def f(self):
     return property(f)
 
 
+def maybe_unwrap_index(obj):
```

Comment:
this is a general function no? if you simply check for a ``__eadata`` attribute. let's put this elsewhere, maybe pandas.core.base

Reply:
The idea is for _eadata to only exist for a few days until we're ready to complete the switchover.  Regardless, maybe_unwrap_index should only be relevant for DTI/TDI/PI.